packages/devtools_app/lib/src/shared/object_tree.dart

```diff
@@ -92,45 +93,106 @@ Future<void> buildVariablesTree(
     }
   }
 
-  if (variable.childCount > DartObjectNode.MAX_CHILDREN_IN_GROUPING) {
-    final numChildrenInGrouping =
-        variable.childCount >= pow(DartObjectNode.MAX_CHILDREN_IN_GROUPING, 2)
-            ? (roundToNearestPow10(variable.childCount) /
-                    DartObjectNode.MAX_CHILDREN_IN_GROUPING)
-                .floor()
-            : DartObjectNode.MAX_CHILDREN_IN_GROUPING;
-
-    var start = variable.offset;
-    final end = start + variable.childCount;
-    while (start < end) {
-      final count = min(end - start, numChildrenInGrouping);
-      variable.addChild(
-        DartObjectNode.grouping(variable.ref, offset: start, count: count),
-      );
-      start += count;
-    }
-  } else if (instanceRef != null && serviceManager.service != null) {
-    final variableId = variable.ref!.isolateRef!.id!;
-    try {
+  try {
+    if (variable.childCount > DartObjectNode.MAX_CHILDREN_IN_GROUPING) {
+      final numChildrenInGrouping =
+          variable.childCount >= pow(DartObjectNode.MAX_CHILDREN_IN_GROUPING, 2)
+              ? (roundToNearestPow10(variable.childCount) /
+                      DartObjectNode.MAX_CHILDREN_IN_GROUPING)
+                  .floor()
+              : DartObjectNode.MAX_CHILDREN_IN_GROUPING;
+
+      var start = variable.offset;
+      final end = start + variable.childCount;
+      while (start < end) {
+        final count = min(end - start, numChildrenInGrouping);
+        variable.addChild(
+          DartObjectNode.grouping(variable.ref, offset: start, count: count),
+        );
+        start += count;
+      }
+    } else if (instanceRef != null && serviceManager.service != null) {
+      final variableId = variable.ref!.isolateRef!.id!;
       final dynamic result = await serviceManager.service!.getObject(
         variableId,
         instanceRef.id!,
         offset: variable.offset,
         count: variable.childCount,
       );
       if (result is Instance) {
-        if (result.associations != null) {
-          variable.addAllChildren(
-            _createVariablesForAssociations(result, isolateRef),
-          );
-        } else if (result.elements != null) {
-          variable
-              .addAllChildren(_createVariablesForElements(result, isolateRef));
-        } else if (result.bytes != null) {
-          variable.addAllChildren(_createVariablesForBytes(result, isolateRef));
-          // Check fields last, as all instanceRefs may have a non-null fields
-          // with no entries.
-        } else if (result.fields != null) {
+        switch (result.kind!) {
```

Comment:
I'm nervous that this will throw if `kind` is `null`.  Do we need the `!`? I think if we remove it we should hit the `default` case instead of throwing an error.

Reply:
`kind` should never be null if `result is Instance`, but I don't think the assertion is necessary if we'll just fall through to the default case anyway.